serveradmin/resources/views.py

```diff
@@ -113,19 +113,24 @@ def index(request):
     per_page = int(request.GET.get(
         'per_page', request.session.get('resources_per_page', 8)))
     request.session['resources_per_page'] = per_page
-    hosts_pager = Paginator(list(hosts.values()), per_page)
+
+    try:
+        hosts_pager = Paginator(list(hosts.values()), per_page).page(page)
```

Comment:
Hm, also possible. I was rather thinking of just setting `page` and `per_page` to a default as well when it is not an integer and negative.

Reply:
A max limit might also make sense, to not make too big queries, if you can already define a max here.

Reply:
> Hm, also possible. I was rather thinking of just setting `page` and `per_page` to a default as well when it is not an integer and negative.

I see your point but I think if a user somewhat entered invalid input here there are only 2 cases.

First one: 
There is a bug in the UI and we should fix it in a way the user can not.

Second one:
The user entered invalid data on purpose and we should make this clear to him by showing an error and notify a developer (by exception).

I don't like the idea of not telling the user that the input was wrong and instead behaving in a way one would not expect or would not match the input but the default.

> A max limit might also make sense, to not make too big queries, if you can already define a max here.

I don't have any it would be arbitrary we have seen 100 hosts working just fine.

Reply:
Yeah, ok, I agree. Maybe let me rephrase the first comment. An invalid `per_page` and `page` should not break or do unwanted things with the server. So, rather an awareness of checking if it is necessary to validate it or validate it early and exit early.

For the `max`, right now it would not make it any better because of the usage of `Query` for pagination, right? I would still suggest to already implement it, so it will just work when tackling the `Query` thing. As we are talking about a absolute max, we can go higher than the average of working hosts. So, I don't see anything against using 1000 as well.